httpcore5/src/test/java/org/apache/hc/core5/net/TestURIAuthority.java

```diff
@@ -220,4 +220,27 @@ public void testCreateFromString() throws Exception {
         }
     }
 
+    @Test
+    public void testCreateFromIPv6String() throws Exception {
+        Assert.assertEquals(new URIAuthority("::1", 8080), URIAuthority.create("[::1]:8080"));
+        Assert.assertEquals(new URIAuthority("::1", -1), URIAuthority.create("[::1]"));
+        try {
+            URIAuthority.create("::1");
```

Comment:
I'm not sure that this case should fail -- we allow hostnames without ports to succeed `URIAuthority.create("hostname")`, so we may want to allow raw ipv6 addresses to succeed as well. On the other hand, if a port is present, it requires the address to be bracketed to avoid ambiguity, which may result in non-obvious failures depending where the value is coming from.

Reply:
This input is ambiguous. Isn't it is allowed to contain ports?

Reply:
Yes, however we define the contract for `URIAuthority.create` so we could say:
1. If the input string is a valid IPv6 address, we assume the full contents represent that address with the default port `-1`
2. Otherwise brackets must be used when a port is present

But that would too complicated for users to understand (or us to maintain), so you're right, it should stay as is.

Reply:
Why isn't this failing?

Reply:
It is failing, we're validating that it throws a`URISyntaxException` because the input string is ambiguous.

Reply:
Right, stupid me.